pkg/cloudprovider/provider/gce/service.go

```diff
@@ -54,7 +55,7 @@ type service struct {
 
 // connectComputeService establishes a service connection to the Compute Engine.
 func connectComputeService(cfg *config) (*service, error) {
-	client := cfg.jwtConfig.Client(context.Background())
+	client := oauth2.NewClient(context.Background(), cfg.clientConfig.TokenSource)
```

Comment:
Nit: technically, it should always exist, but `cfg.clientConfig` is a pointer and it can NPE if it's not set. Maybe it's worth checking for that before using `TokenSource`.

Reply:
As you mentioned, pointer is already initialised before any use of this field thus what would the check help with?

Reply:
The idea is if we eventually call this function before initializing the struct or we introduce some bug, we can log it instead of NPE. No strong opinions here, so we can ignore this.